### PR TITLE
Draft 22

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
@bifurcation This doesn't implement the compat mode stuff and will fail on ChangeCipherSpec. I plan to submit that as a separate PR. LMK if you want to land those together.

File: common.go [hunk 1]
```diff
@@ -22,6 +21,9 @@ const (
 	RecordTypeApplicationData RecordType = 23
 )
 
+const tls10Version uint16 = 0x0301
+const tls12Version uint16 = 0x0303
```
Comment (bifurcation) [inline common.go]: Suggest changing the `var` block above to `const` and just adding these up there.

File: handshake-messages.go [hunk 2]
```diff
@@ -28,9 +28,10 @@ type ClientHelloBody struct {
 	// Omitted: clientVersion
 	// Omitted: legacySessionID
```
Comment (bifurcation) [inline handshake-messages.go]: Please remove this comment line, since we now have `LegacySessionID`

File: handshake-messages.go [hunk 3]
```diff
@@ -150,10 +152,12 @@ func (hrr *HelloRetryRequestBody) Unmarshal(data []byte) (int, error) {
 //     Extension extensions<0..2^16-1>;
 // } ServerHello;
 type ServerHelloBody struct {
-	Version     uint16
-	Random      [32]byte
-	CipherSuite CipherSuite
-	Extensions  ExtensionList `tls:"head=2"`
+	Version                 uint16
+	Random                  [32]byte
+	LegacySessionID         []byte `tls:"head=1,max=32"`
```
Comment (bifurcation) [inline handshake-messages.go]: Please update the comment above with the new ServerHello structure.

**Pull request merged**